### PR TITLE
Add optional span param to end_span

### PR DESCRIPTION
**Pull request by matheussilvasantos**
## What does this pull request do?

Add optional span param to end_span

## Why is it important?

When using the gem in asynchronous systems, multiple spans will be created in parallel, and they aren't going to finish in the same order they were created. Therefore, to have spans with the correct duration, we need to pass the span we want to finish - in this case, it's the developer responsibility to hold the span created and pass it to `end_span`.

Usage example:

```ruby
require "eventmachine"
require "em-http"
require "elastic_apm"

puts "Starting..."

ElasticAPM.start

EM.run do
  ElasticAPM.start_transaction("Transaction")

  count = 0

  ["https://example.com", "https://example.org"].each do |url|
    span = ElasticAPM.start_span("GET #{url}")

    http = EventMachine::HttpRequest.new(url).get

    http.errback do
      ElasticAPM.end_span(span)
      count += 1
    end

    http.callback do
      ElasticAPM.end_span(span)
      count += 1
    end
  end

  timer = EventMachine::PeriodicTimer.new(1) do
    if count == 2
      ElasticAPM.end_transaction
      puts "Stopping..."
      EM.add_timer(5) do
        ElasticAPM.stop
        EM.stop
      end
      timer.cancel
    end
  end
end
```

## Checklist

- [x] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/). 
- [x] My code follows the style guidelines of this project (See `.rubocop.yml`)
- [x] I have rebased my changes on top of the latest master branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing [**unit** tests](https://github.com/elastic/apm-agent-ruby/blob/master/CONTRIBUTING.md#testing) pass locally with my changes
- [x] I have made corresponding changes to the documentation
- [x] I have updated [CHANGELOG.asciidoc](CHANGELOG.asciidoc)
- [x] I have updated [supported-technologies.asciidoc](docs/supported-technologies.asciidoc)
- [ ] Added an API method or config option? Document in which version this will be introduced
(I'm not sure on this one)

### CODE REVIEW

File: lib/elastic_apm/instrumenter.rb [hunk 1]
```diff
@@ -222,8 +222,14 @@ def start_span(
...
     # rubocop:enable Metrics/PerceivedComplexity
 
-    def end_span
-      return unless (span = current_spans.pop)
+    def end_span(span = nil)
+      unless span.nil?
+        current_spans.delete(span)
+      else
+        span = current_spans.pop
+      end
```
Comment (mikker) [inline lib/elastic_apm/instrumenter.rb:230]:
```suggestion
      if span
        current_spans.delete(span)
      else
        span = current_spans.pop
      end
```

**Review by mikker — APPROVED** [review]
This is wonderful! Great addition. I don't remember anyone else using the agent with EventMachine so keep the fixes and additions coming as you encounter them ❤️

**Comment by mikker** [comment]
jenkins, run the tests

File: lib/elastic_apm/instrumenter.rb (hunk 1 above)
Comment (matheussilvasantos) [reply lib/elastic_apm/instrumenter.rb]: Updated.

**Comment by mikker** [comment]
jenkins, run the tests

**Comment by estolfo** [comment]
Thanks @matheussilvasantos!

**Pull request merged**